src/config/config.service.ts

```diff
@@ -0,0 +1,44 @@
+import * as Joi from 'joi';
+import * as fs from 'fs';
+import * as dotenv from 'dotenv'
+import {Injectable} from "@nestjs/common";
+
+export interface EnvConfig {
+    [key: string]: string;
+}
+
+@Injectable()
+export class ConfigService {
+    private readonly envConfig: EnvConfig;
+
+    constructor(filePath: string) {
+        const config = dotenv.parse(fs.readFileSync(filePath));
+        this.envConfig = this.validateInput(config);
+    }
+
+    /**
+     * Ensures all needed variables are set, and returns the validated JavaScript object
+     * including the applied default values.
+     */
+    private validateInput(envConfig: EnvConfig): EnvConfig {
+        const envVarsSchema: Joi.ObjectSchema = Joi.object({
+            NODE_ENV: Joi.string()
+                .valid(['development', 'production', 'test', 'provision'])
+                .default('development'),
+            PORT: Joi.number().default(3000),
+        });
+
+        const {error, value: validatedEnvConfig} = Joi.validate(
+            envConfig,
+            envVarsSchema,
+        );
+        if (error) {
+            throw new Error(`Config validation error: ${error.message}`);
+        }
+        return validatedEnvConfig;
+    }
+
+    get(key: string): string {
+        return this.envConfig[key];
+    }
```

Comment:
Jak masz wolę i ochotę ułatwić sobie typowane życie, to możesz zamienić generyczny `string` na coś bardziej precyzyjnego (jako typ parametru i zwrotka). Musiałbyś jednak otypować `EnvConfig`, co samo w sobie też jest dobrym pomysłem (doda to warstwę dokumentacji konfiguracji).

```
interface EnvConfig {
  name: string
  age: number
}

class ConfigService {
  get<K extends keyof EnvConfig>(key: K): EnvConfig[K] {
    return this.envConfig[key];
  }
}
```